torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -373,14 +373,33 @@ void ProcessGroupAgent::enqueueRecv(RecvWork work) {
   threadPool_.run(std::bind(
       [&](RecvWork& work) {
         torch::Tensor& payload = work.payload_;
-        auto data = wireDeserialize(payload.storage().data(), payload.numel());
-        Message message(
-            std::move(data.first),
-            std::move(data.second),
-            work.type_,
-            work.id_);
+
+        // Deserialize incoming message.
+        Message message;
+        try {
+          auto data =
+              wireDeserialize(payload.storage().data(), payload.numel());
+          message = std::move(Message(
+              std::move(data.first),
+              std::move(data.second),
+              work.type_,
+              work.id_));
+        } catch (std::exception& e) {
+          LOG(ERROR) << "Could not deserialize " << payload.numel()
+                     << " byte payload from " << work.from_.id_;
```

Comment:
Is it worth it to try sending a response back to `work.from_id` indicating that it's message couldn't be deserialized?

Reply:
Good idea, seems like it would be more reasonable to send a relevant exception back.